p2p/src/net/mock/transport/traits.rs

```diff
@@ -46,11 +47,14 @@ pub trait MockTransport: Send + 'static {
     /// A messages stream.
     type Stream: MockStream;
```

Comment:
I'm having trouble understanding why StreamKey can't be just part of Stream... something doesn't add up.

Reply:
Moved to `MockStream`. Though referencing `StreamKey` become a bit convoluted ("ambiguous associated type" error):

```
-        stream_key: T::StreamKey,
+        stream_key: <<T as MockTransport>::Stream as MockStream>::StreamKey,
```

Reply:
This is now irrelevant after the changes. Resolving.